reports/tasks.py

```diff
@@ -76,6 +66,22 @@ def send_trial_instances_report(recipients=settings.TRIAL_INSTANCES_REPORT_RECIP
     end_of_last_month = (beginning_of_this_month - timedelta(days=1))
     beginning_of_last_month = end_of_last_month.replace(day=1)
 
+    # Get instances with an active betatestapplication,
+    # whose InstanceReference has an active OpenEdXAppServer
+    # and was created last month
+    instances = OpenEdXInstance.objects.exclude(
+        betatestapplication__isnull=True
+    ).filter(
+        ref_set__openedxappserver_set___is_active=True
+    ).filter(
+        ref_set__created__range=[
+            beginning_of_last_month.strftime("%Y-%m-%d"),
+            end_of_last_month.strftime("%Y-%m-%d")
```

Comment:
@spokerman12, this has the off-by-one error as mentioned [here](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#range). Can you fix it and also write a test case for this scenario?

Reply:
Thank you for picking up this case. I've modified the filter and added the unit test